.github/workflows/pyfiction-docstring-generator.yml

```diff
@@ -20,6 +16,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+          fetch-depth: 0 # Fetch all history for all branches and tags
```

Comment:
Why is this important?

Reply:
Previously, the docstring generator only worked for braches forked directly from `fiction` and not, e.g., `Drewniok/fiction`, which lead to this error message:

Error: Error: fatal: You are not currently on a branch.
  To push the history leading to the current (detached HEAD)
  state now, use
  
      git push origin HEAD:<name-of-remote-branch>

Reply:
Understood. Many thanks!